### PR TITLE
[Refactor] data flow

### PR DESCRIPTION
**Pull request by MeowZheng**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

1. As title

## Modification

WIP

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmseg/models/segmentors/base.py [hunk 1]
```diff
@@ -130,11 +128,12 @@ def aug_test(self, batch_inputs, batch_img_metas):
...
 
-    def postprocess_result(self, seg_logits_list: List[dict],
-                           batch_img_metas: List[dict]) -> list:
+    def postprocess_result(self,
+                           seg_logits: Tensor,
+                           data_samples: List[dict] = None) -> list:
         """ Convert results list to `SegDataSample`.
         Args:
-            seg_logits_list (List[dict]): List of segmentation results,
+            seg_logits (Tensor): List of segmentation results,
```
Comment (Zachary-66) [inline mmseg/models/segmentors/base.py:136]: The comment of `seg_logits` needs to be updated

File: mmseg/engine/hooks/visualization_hook.py [hunk 2]
```diff
@@ -66,15 +66,15 @@ def __init__(self,
...
+                    data_batch: dict,
                     outputs: Sequence[SegDataSample],
                     mode: str = 'val') -> None:
         """Run after every ``self.interval`` validation iterations.
 
         Args:
             runner (:obj:`Runner`): The runner of the validation process.
             batch_idx (int): The index of the current batch in the val loop.
-            data_batch (Sequence[dict]): Data from dataloader.
+            data_batch dict: Data from dataloader.
```
Comment (HAOCHENYE) [inline mmseg/engine/hooks/visualization_hook.py:77]:
```suggestion
            data_batch (dict): Data from dataloader.
```

File: mmseg/models/segmentors/base.py [hunk 3]
```diff
@@ -130,13 +128,13 @@ def aug_test(self, batch_inputs, batch_img_metas):
         """Placeholder for augmentation test."""
         pass
 
-    def postprocess_result(self, seg_logits_list: List[dict],
-                           batch_img_metas: List[dict]) -> list:
+    def postprocess_result(self,
+                           seg_logits: Tensor,
+                           data_samples: List[dict] = None) -> list:
```
Comment (HAOCHENYE) [inline mmseg/models/segmentors/base.py:133]:
```suggestion
                           data_samples: Optional[List[dict]] = None) -> list:
```

**Review by HAOCHENYE — COMMENTED** [review]
Should demo adjusts with new data_flow？

**Review by xiexinch — APPROVED** [review]

**Pull request merged**